java/client-test/src/test/java/no/bankaxept/epayment/test/client/base/AccessTokenPublisherTest.java

```diff
@@ -52,30 +55,44 @@ public void tearDown(){
     @Test
     public void should_schedule_on_startup_then_new_on_success() throws IOException {
         doReturn(new SinglePublisher<>(new HttpResponse(200, tokenResponse()), executor)).when(httpClientMock).post(eq("uri"), any(), any());
-        accessTokenProcessor = new ScheduledAccessTokenPublisher("uri", "key", "username", "password", clock, schedulerMock, httpClientMock);
+        accessTokenProcessor = createPublisher();
         accessTokenProcessor.subscribe(subscriberMock);
         verify(schedulerMock).schedule(any(Runnable.class), eq(0L), eq(TimeUnit.SECONDS));
-        verify(schedulerMock, Mockito.after(2000)).schedule(any(Runnable.class), eq(3000L), eq(TimeUnit.SECONDS));
+        verify(schedulerMock, Mockito.after(2000)).schedule(any(Runnable.class), eq(3590L), eq(TimeUnit.SECONDS));
         verify(subscriberMock).onNext("a-token");
+        accessTokenProcessor.subscribe(anotherSubscriberMock);
+        verify(anotherSubscriberMock).onNext("a-token");
```

Comment:
Can you comment what's going on here so other developers might get it, too?

Reply:
I added a comment here